src/main/java/org/reactome/addlinks/db/ReferenceObjectCache.java

```diff
@@ -106,183 +106,217 @@ private static synchronized void buildReferenceCaches(String className, Map<Stri
-							@SuppressWarnings("unchecked")
-							Collection<GKInstance> referringRefTranscripts = (Collection<GKInstance>) referenceObject.getReferers(ReactomeJavaConstants.referenceTranscript);
-							allSpecies = new HashSet<String>(referringRefTranscripts.size());
-							// Populate a list of species that this ReferenceRNASequence could be cached by.
-							for (GKInstance refTranscript : referringRefTranscripts)
-							{
-								try
-								{
-									allSpecies.add(  ((GKInstance)refTranscript.getAttributeValue(ReactomeJavaConstants.species)).getDBID().toString() );
-								}
-								catch (InvalidAttributeException e)
-								{
-									logger.error("Could not get the \"{}\" attribute for {} because it not valid for this object.", ReactomeJavaConstants.species, refTranscript);
-								}
-								catch (Exception e)
-								{
-									logger.error("Could not get the \"{}\" attribute for {}. Reason: {}", ReactomeJavaConstants.species, refTranscript, e.getMessage());
-								}
-							}
-						}
-						catch (Exception e1)
-						{
-							logger.error("Could not get ReferenceTranscripts for {} (while trying to determine the species)", referenceObject);
-						}
-						
-					}
-				}
-
-				// If the list wasn't initialized yet, it means that we are probably dealing with a ReferenceGeneProduct or a ReferenceDNASequence which should
-				// have its own species.
-				if (allSpecies == null)
-				{
-					allSpecies = new HashSet<String>(1);
-					String species;
-					try
-					{
-						species = String.valueOf( ((GKInstance) referenceObject.getAttributeValue(ReactomeJavaConstants.species)).getDBID() );
-						allSpecies.add(species);
-					}
-					catch (InvalidAttributeException e)
-					{
-						logger.error("Could not get the \"{}\" attribute for {} because it not valid for this object.", ReactomeJavaConstants.species, referenceObject);
-					}
-					catch (Exception e)
-					{
-						logger.error("Could not get the \"{}\" attribute for {}. Reason: {}", ReactomeJavaConstants.species, referenceObject, e.getMessage());
-					}
-					
-				}
-				if (cacheBySpecies != null)
-				{
-					for (String species : allSpecies)
-					{
-						//Add to the Species Cache
-						if (! cacheBySpecies.containsKey(species))
-						{
-							List<GKInstance> bySpecies = Collections.synchronizedList(new LinkedList<GKInstance>());
-							bySpecies.add(referenceObject);
-							cacheBySpecies.put(species, bySpecies);
-						}
-						else
-						{
-							cacheBySpecies.get(species).add(referenceObject);
-						}
-					}
-				}
+				addObjectToSpeciesCache(className, cacheBySpecies, referenceObject);
 			}
 			//ReferenceDatabase Cache
 			//
 			if (cacheByRefDB != null)
 			{
-				String refDBID;
+				addObjectToRefDBCache(cacheByRefDB, referenceObject);
+			}
+			// DB_ID Cache
+			cacheByID.put(String.valueOf(referenceObject.getDBID()), referenceObject);
+			
+			// Identifier cache
+			if ( !className.equals(ReactomeJavaConstants.Reaction) )
```

Comment:
Why is this for `Reaction` and not `ReactionlikeEvent`?

Reply:
We only create references on Reactions for Rhea and TargetPathogen, and I guess when it was discussed, it was decided to use `Reaction`.

Reply:
Also I just realized that if you are in fact trying to determine broadly if its a `ReactionlikeEvent`, rather than specifically a `Reaction`, the `instance.getSchemClass().isa(ReactomeJavaConstants.ReactionlikeEvent`  approach will be needed to get proper coverage